src/commonMain/kotlin/baaahs/show/live/AutoMode.kt

```diff
@@ -0,0 +1,68 @@
+package baaahs.show.live
+
+import baaahs.control.OpenButtonControl
+import baaahs.doRunBlocking
+import baaahs.util.Logger
+import baaahs.util.globalLaunch
+import kotlin.random.Random
+import kotlinx.coroutines.*
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.coroutineContext
+
+class AutoMode(
+    private val openShow: OpenShow
+) {
+
+    var indexSeed = 13;
+    var autoMode = false;
+    var running = false;
+    val isRunning: Boolean get() = run {
+        running
+    }
+
+    val autoModeOn: Boolean get() = run {
+        autoMode
+    }
+
+    var gadgetWizard: Job? = null
+
+    fun start() {
+        if (isRunning) return
+        running = true
+        autoMode = true
+        gadgetWizard = CoroutineScope(Dispatchers.Unconfined).launch {
+            playWithButtons()
+        }
+    }
+
+    var frequency = 5000L
+
+    private suspend fun playWithButtons() {
+        val controls: List<OpenButtonControl> = openShow.allFilterButtonControls
+        logger.info { "Controls " + controls.joinToString { it.gadget.title } }
+        val random = Random(indexSeed++)
```

Comment:
You could store the `Random` as a `private var` if you want, no need really to reinitialize it every time, you'll keep getting pseudo-random numbers forever.

Fwiw I don't usually bother with creating explicit `Random`s unless unit tests interact with it, in which case they're awesome.

Reply:
Yeah not sure how to get truly random results since the seed is always the same, so the auto mode will always pick the same number sequence :/ 

One solution is to get the current time in seconds as the seed to always get diff results, but I couldn't reference this for some reason: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.system/get-time-millis.html